tests/utility/hash_test.cu

```diff
@@ -48,6 +48,63 @@ __host__ __device__ bool check_hash_result(typename Hash::argument_type const& k
   return (h(key) == expected);
 }
 
+// Overload for hash functions without a seed
+template <typename Hash>
+__host__ __device__ bool check_hash_result(typename Hash::argument_type const& key,
+                                           typename Hash::result_type expected) noexcept
+{
+  Hash h;
+  return (h(key) == expected);
+}
+
+template <typename OutputIter>
+__global__ void check_identity_hash_result_kernel(OutputIter result)
+{
+  int i = 0;
+
+  result[i++] = check_hash_result<cuco::identityhash<signed char>>(SCHAR_MIN, SCHAR_MIN);
+  result[i++] = check_hash_result<cuco::identityhash<signed char>>(SCHAR_MAX, SCHAR_MAX);
+
+  result[i++] = check_hash_result<cuco::identityhash<int32_t>>(INT32_MIN, INT32_MIN);
+  result[i++] = check_hash_result<cuco::identityhash<int32_t>>(INT32_MAX, INT32_MAX);
+
+  result[i++] = check_hash_result<cuco::identityhash<int64_t>>(INT64_MIN, INT64_MIN);
+  result[i++] = check_hash_result<cuco::identityhash<int64_t>>(INT64_MAX, INT64_MAX);
+
+#if defined(CUCO_HAS_INT128)
+  result[i++] = check_hash_result<cuco::identityhash<__int128>>((__int128)INT64_MAX + 1,
+                                                                (__int128)INT64_MAX + 1);
+#endif
+}
+
+TEST_CASE("Test cuco::identityhash", "")
+{
+  SECTION("Check if host-generated hash values match the identity function.")
+  {
+    CHECK(check_hash_result<cuco::identityhash<signed char>>(SCHAR_MIN, SCHAR_MIN));
+    CHECK(check_hash_result<cuco::identityhash<signed char>>(SCHAR_MAX, SCHAR_MAX));
+
+    CHECK(check_hash_result<cuco::identityhash<int32_t>>(INT32_MIN, INT32_MIN));
+    CHECK(check_hash_result<cuco::identityhash<int32_t>>(INT32_MAX, INT32_MAX));
+
+    CHECK(check_hash_result<cuco::identityhash<int64_t>>(INT64_MIN, INT64_MIN));
+    CHECK(check_hash_result<cuco::identityhash<int64_t>>(INT64_MAX, INT64_MAX));
+
+#if defined(CUCO_HAS_INT128)
+    CHECK(check_hash_result<cuco::identityhash<__int128>>((__int128)INT64_MAX + 1,
+                                                          (__int128)INT64_MAX + 1));
+#endif
+  }
+  SECTION("Check if device-generated hash values match the identity function.")
+  {
+    thrust::device_vector<bool> result(7, true);
+
+    check_identity_hash_result_kernel<<<1, 1>>>(result.begin());
+
+    CHECK(cuco::test::all_of(result.begin(), result.end(), [] __device__(bool v) { return v; }));
```

Comment:
```suggestion
    CHECK(cuco::test::all_of(result.begin(), result.end(), thrust::identity<bool>{}));
```
nit: a perfect use case for identity.

Reply:
Playing the devil's advocate for a moment: When switching to an alias typedef as proposed earlier in this PR discussion, do we even need a unit test for it?

I don't want to be the guy who proposes having fewer unit tests in our project (hell no!), but this seems like a test that just checks if `thrust::identity` works as advertised, which should already be tested upstream, i.e., in Thrust itself.

Reply:
I agree that `thrust::identity` works perfectly here, however `[] __device__(bool v) { return v; }` is used by the existing tests in this file for `xxhash` and `mumurhash`. See lines 168 and 343. Should we leave this as-is to maintain consistency? Changing the tests for `xxhash` and `murmurhash` to `thrust::identity` seems out of scope.

As for @sleeepyjack 's comment; even if we change the current code to an alias typedef, in the future I don't think `IdentityHash` will remain that way. Does it make sense to keep the unit tests to protect the functionality of `IdentityHash` against future changes?

Reply:
It's fine. Another PR moving all device lambda to identity is more than welcome

Reply:
To expand on the second half of [what @njones93531 said](https://github.com/NVIDIA/cuCollections/pull/514/files#r1652738321), as I described [here](https://github.com/NVIDIA/cuCollections/pull/505#discussion_r1646605810), it will greatly improve the safety and usability of cuco perfect hashing if we design the "perfect probing" scheme to require that the hash function used with it is perfect.  That is, we want a compilation failure if the user tried to use a non-perfect hash with perfect probing because it will definitely cause errors in the hash table's function.

The way that I propose to get that safety is like [how type traits work](https://www.internalpointers.com/post/quick-primer-type-traits-modern-cpp).  We'll define a method `bool is_perfect()` of hash functors that are perfect.  And the perfect probing scheme will fail at compile-time if you try to use it with a hash functor that doesn't have that function or the function returns `false`.

In order to do this, you'll have to have an `identity_hash` like above, that extends the `thrust::identity` and adds the `is_perfect` function into it.

So, while we *don't* need the file `identity_hash.cuh` at the moment, we *will* need it later.  We can add it now, like this PR does, or remove this and add it later along with the perfect hashing.

@njones93531 Let's not [bikeshed](https://en.wikipedia.org/wiki/Law_of_triviality) this one!  Let @PointKernel read this argument, decide if he wants the class now or later, and we'll just do it.  :smile: